libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationView.kt

```diff
@@ -143,46 +180,141 @@ class NavigationView : ConstraintLayout, LifecycleObserver {
+                    merge(
+                        flowOf(
+                            SpeedLimitAction.UpdateNavigationState(state)
+                        ),
+                        flowOf(
+                            SpeedLimitAction.UpdateVisibility(state)
+                        )
+                    )
+                )
+                maneuverViewModel.consumeAction(
+                    merge(
+                        flowOf(
+                            ManeuversAction.UpdateNavigationState(state)
+                        ),
+                        flowOf(
+                            ManeuversAction.UpdateVisibility(state)
+                        )
+                    )
+                )
+                volumeViewModel.consumeAction(
+                    merge(
+                        flowOf(
+                            SoundButtonAction.UpdateNavigationState(state)
+                        ),
+                        flowOf(
+                            SoundButtonAction.UpdateVisibility(state)
+                        )
+                    )
+                )
+                tripProgressViewModel.consumeAction(
+                    merge(
+                        flowOf(
+                            TripProgressAction.UpdateNavigationState(state)
+                        ),
+                        flowOf(
+                            TripProgressAction.UpdateVisibility(state)
+                        )
+                    )
+                )
+                recenterViewModel.consumeAction(
+                    merge(
+                        flowOf(
+                            RecenterButtonAction.UpdateNavigationState(state)
+                        ),
+                        flowOf(
+                            RecenterButtonAction.UpdateVisibility(
+                                state,
+                                NavigationCameraState.IDLE
+                            )
+                        )
+                    )
+                )
+                routeOverviewViewModel.consumeAction(
+                    merge(
+                        flowOf(
+                            RouteOverviewButtonAction.UpdateNavigationState(state)
+                        ),
+                        flowOf(
+                            RouteOverviewButtonAction.UpdateVisibility(
+                                state,
+                                NavigationCameraState.IDLE
+                            )
+                        )
+                    )
+                )
             }
         }
         lifeCycleOwner.lifecycleScope.launch {
-            containerVisibilityViewModel.containerVisibilityState().collect { state ->
-                renderVisibility(state)
+            speedLimitViewModel.speedLimitState().collect { state ->
+                binding.speedLimitContainer.visibility = state.isVisible.toVisibility()
+                (speedLimitView as MapboxSpeedLimitView).render(state.speedLimit)
```

Comment:
We'll need to have a more extensive logic here and only run an update if the view actually is an instance of `MapboxSpeedLimitView` (same for all other views). If someone would use a custom view for `speedLimitView`, we'd crash here.

If it's not our view, we shouldn't even subscribe to the view model for any updates. We need a controlling unit that understands the configuration and which views are ours and which aren't so that we can skip them.

Reply:
Let's also try to move it out of the `NavigationView` class so that it doesn't grow.

Reply:
> If it's not our view, we shouldn't even subscribe to the view model for any updates. We need a controlling unit that understands the configuration and which views are ours and which aren't so that we can skip them.

I agree with this totally and was already planning to cut a ticket on this task. I will remove this line here in this PR as this might need some thoughts on how to do even though I have something in mind. But, let's not block this PR and introduce a proper system for it separately.

> Let's also try to move it out of the NavigationView class so that it doesn't grow.

Do you mean all `*.collect { }` calls as there are several

/cc @LukasPaczos